File: algorithms/searching/linear_search/linear.cpp
#include <iostream>
using namespace std;

int linear_search(int numbers[], int size, int target)
{

    for (int i = 0; i < size; i++)
    {
        if (numbers[i] == target)
        {
            return i;
        }
    }
    return -1;
}

int main()
{
    int numbers[] = {2, 3, 4, 10, 40};
    int target = 10;
    int size = sizeof(numbers) / sizeof(numbers[0]);
    int result = linear_search(numbers, size, target);
    if (result == -1)
    {
        cout << "Element doesn't exist!" << endl;
    }
    else
    {
        cout << "Element is present at index " << result << endl;
    }
    return 0;
}

File: algorithms/searching/linear_search/linear.py
def linear_search(numbers: list, target: int) -> int:
    for i in range(len(numbers)):
        if numbers[i] == target:
            return i
    return -1

def main():
    numbers = [2, 3, 4, 10, 40]
    target = 10
    result = linear_search(numbers, target)
    if result == -1:
        print("The element doesn't exist :(")
    else:
        print(f"{target} exists at index {result}!")

if __name__ == "__main__":
    main()